### PR TITLE
rpc: avoid use of cgo by hard-coding maxPathSize

### PR DESCRIPTION
**Pull request by zchee**
Avoid using `cgo` with `go tool cgo -godefs`.

Generated by:

```console
$ docker container run --rm -it --entrypoint=sh -v $PWD:/go/src/github.com/ethereum/go-ethereum -w /go/src/github.com/ethereum/go-ethereum golang:bullseye -c 'make godef'
```

### CODE REVIEW

**Comment by zchee** [comment]
If you want to get values per `GOOS`, commit those files too.

**Comment by fjl** [comment]
Hmm, we already have a version of this file without cgo, `constants_unix_nocgo.go`

**Comment by zchee** [comment]
@fjl But document said:

> On Linux, sun_path is 108 bytes in size

However, at least bullseye distro, have different size (`0x6e`) .
I think this change will get more realstic value.

**Comment by zchee** [comment]
ah, I'm misunderstood. Fix later

**Comment by zchee** [comment]
@fjl PTAL.

**Comment by holiman** [comment]
Sorry, I don't understand. I am not familiar with `godefs`.  What is the problem that you want to solve with this PR?

**Comment by fjl** [comment]
This PR removes the dependency on cgo by generating the `max_path_size` constant using the `godefs` tool. We use this constant to validate the length of IPC file names. 

If the dependency on cgo is a problem, we could also just remove the `constants_unix.go` file and use our *existing* non-cgo alternative for all platforms.

**Comment by zchee** [comment]
@fjl 
Rename `constants_unix.go` to `constants.go`. PTAL

File: rpc/ipc_unix.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ import (
 // ipcListen will create a Unix socket on the given endpoint.
 func ipcListen(endpoint string) (net.Listener, error) {
 	// account for null-terminator too
-	if len(endpoint)+1 > int(max_path_size) {
-		log.Warn(fmt.Sprintf("The ipc endpoint is longer than %d characters. ", max_path_size-1),
+	if len(endpoint)+1 > maxPathSize {
```
Comment (holiman) [inline rpc/ipc_unix.go:35]: If we want to make things simpler, why not just put the constant in this file and get rid of `constants.go` too?
We can just  
```suggestion
	//  On Linux, sun_path is 108 bytes in size
	// see http://man7.org/linux/man-pages/man7/unix.7.html
	var maxPathSize = int(108)
	if len(endpoint)+1 > maxPathSize {
```

**Comment by zchee** [comment]
@fjl Thanks!

**Pull request merged**